### ISSUE TITLE
retraining diffpack

### ISSUE THREAD

**Issue by linbc-73** [DeepGraphLearning/DiffPack#6]
Hi, I am very interested in Diffpack and hoping to know if there is any script that I can retrain Diffpack with?

**Comment by zytzrh** [comment]
Hi @linbc-73, thanks for pointing this out. We're working on reformating and updating the training code, and we expect to have this update ready within the next month.

**Comment by gxglxy** [comment]
Hi, I am also interested in DiffPack and wonder when would you release the training code. Thank you for your hard work.

**Comment by s1234533** [comment]
+1

**Comment by Oxer11** [comment]
Hi, sorry for the delay! I've written a demo training code and put it in the branch (https://github.com/DeepGraphLearning/DiffPack/tree/training). You can use this for reference. The official version will be released soon.

**Comment by linbc-73** [comment]
Hi
I still have several questions about training DiffPack. After training 400 epochs using the script you provided in the training branch with the same configuration, I got unexpected chi4(chi3 as in the codes) deg results on the validation set. The chi4(chi3 as in the codes) deg was around 90 through the whole time while the deg for other chi id decreased. Then I  printed the loss for each chi id during training and found the loss for chi4(chi3 as in the codes) fail to converge. 
<img width="1234" alt="截屏2024-03-10 11 52 05" src="https://github.com/DeepGraphLearning/DiffPack/assets/83945633/a4017423-fbe1-4a7d-8831-0675050d3cb1">
<img width="1249" alt="截屏2024-03-10 11 53 45" src="https://github.com/DeepGraphLearning/DiffPack/assets/83945633/9e8076ab-9219-4940-b4f7-c68ce892efa7">

There could be several reasons that lead to the problem and I am hoping you could help me get some clues about that.
1. The data I used for training could have some mistake, I collected the protein data from bc40 by myself and changed the codes for data loading. Is it possible that you could share the .pkl format data that you used with me? So that I can make sure there is nothing wrong with my data.
2. I changed a few places in the training branch. Except the data loading process, I changed line 181-182 of generate() of class TorsionDiffusion in /diffpack/task.py as the dimension of pred_score had been incorrect. And I wonder if there are other little mistakes in the codes that could cause upper-mentioned problems.
3. During debugging, I also checked using teacher-forcing during inference, which means I set the atoms of chi1-chi-i in ground-truth positions when generating chi-(i+1). And I found the deg of all 4 chi dihedral to be almost zero. So I guess the problem could also resulted from the inconsistency between training and testing because of teacher-forcing training. But it doesn't explain why the diffusion loss of chi-3 didn't converge.
And I noticed there is a comment mark says rotamer.set_chis() might have bugs in line 108 task.py, and I used this function when I used teacher-forcing in inference.
Thanks for your help in advance!
